### ISSUE TITLE
Replace pylint with flake8 plugins

### ISSUE THREAD

**Issue by santisoler** [fatiando/harmonica#284]
We've been using `pylint` for a long time to help us improve the quality of our code.
Nevertheless, it's not new that meeting all its requirements is a very demanding task: some of its complains are very useful (unused imports, unused variables, too long functions, etc), but there are a lot more that we usually decide to ignore. The amount of ignored complains is not small: I think we need to add  `pylint: disable=...` comments on every PR we open. See #283 as a recent example.

So it might be better to go with a better alternative: one that is faster to run, less strict and that allows us to control which types of check we want to run and which don't.
A good alternative would be to use `flake8` extensions that take care of specific checks.

I'll leave a list of possible candidates, so feel free to add any extensions you think useful:

- [flake8-bugbear](https://github.com/PyCQA/flake8-bugbear): find likely bugs and design problems in your program.
- [flake8-simplify](https://github.com/MartinThoma/flake8-simplify): simplify your code.
- [flake8-variables-names](https://github.com/best-doctor/flake8-variables-names): make more readable variables names.
- [flake8-functions](https://github.com/best-doctor/flake8-functions): checks functions length, number of arguments, etc.
- [flake8-builtins](https://github.com/gforcada/flake8-builtins): Check for python builtins being used as variables or parameters.
- [flake8-comprehensions](https://github.com/adamchainz/flake8-comprehensions): write better list/set/dict comprehensions.
- [pep8-naming](https://github.com/PyCQA/pep8-naming): Check your code against PEP 8 naming conventions.
- [flake8-mutable](https://github.com/ebeweber/flake8-mutable): check against using mutable values as default arguments.
- [flake8-rst-docstrings](https://github.com/peterjc/flake8-rst-docstrings): validating Python docstrings markup as reStructuredText (RST)

Discarded:
- ~[darglint](https://github.com/terrencepreilly/darglint): checks whether a docstring's description matches the actual function/method implementation~
    - (takes too much time to run)
- ~[flake8-tidy-imports](https://github.com/adamchainz/flake8-tidy-imports): write tidier imports.~
    - (not available through conda)
- ~[flake8-isort](https://github.com/gforcada/flake8-isort): Use `isort` to check if the imports on your python files are sorted the way you expect.~
    - (we will use `isort`, but we don't need to list all its complains, we just want it to solve them)
- ~[flake8-black](https://github.com/peterjc/flake8-black): Run `black --check` from within `flake8` ecosystem.~
    - (same as for `flake8-isort`)
- ~[flake8-functions-names](https://github.com/Melevir/flake8-functions-names): provide better naming for functions.~
    - (doesn't offer major features)
- ~[flake8-requirements](https://github.com/Arkq/flake8-requirements): checks/validates package import requirements. It reports missing and/or not used project direct dependencies.~
    - (not available through conda)
- ~[flake8-eradicate](https://github.com/wemake-services/flake8-eradicate): find commented out (or so called "dead") code.~
    - (the version available through conda is olddated)
- ~[flake8-unused-arguments](https://github.com/nhoad/flake8-unused-arguments)~
    - (not available through conda)

The [awesome-flake8-extensions](https://github.com/DmytroLitvinov/awesome-flake8-extensions) is a good place for looking for `flake8` extensions.

**Comment by leouieda** [comment]
Maybe this one as well: https://github.com/PyCQA/flake8-bugbear

**Comment by santisoler** [comment]
After the [2021-11-19 Developer Meeting](https://github.com/fatiando/community/blob/main/development-calls/2021.md#2021-11-19) we decided not to include any flake8 extension that isn't available through conda.
Besides, we decided that `isort` is a good tool to incorporate, although we really don't want to include `flake8-isort`: since `isort` is capable of solving their own issues, a simple `make format` would solve them, without the need of `flake8` to list them all.
The same goes for `flake8-black`.
